.github/files/update-project-mirrors.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+git_setup()
+{
+  cat <<- EOF > "$HOME/.netrc"
+		machine github.com
+			login matticbot
+			password $GITHUB_TOKEN
+		machine api.github.com
+			login matticbot
+			password $GITHUB_TOKEN
+EOF
+  chmod 600 "$HOME/.netrc"
+
+  git config --global user.email "matticbot@users.noreply.github.com"
+  git config --global user.name "matticbot"
+}
+
+# Halt on error
+set -eo pipefail
+
+if [[ -n "$CI" ]]; then
+	git_setup
+fi
+
+BASE=$(pwd)
+CLONE_BASE=$(mktemp -d "${TMPDIR:-/tmp}/jetpack-update-project-mirrors.XXXXXXXX")
+MONOREPO_COMMIT_MESSAGE=$(git show -s --format=%B $GITHUB_SHA)
+COMMIT_MESSAGE=$( echo "${MONOREPO_COMMIT_MESSAGE}\n\nCommitted via a GitHub action: https://github.com/automattic/jetpack/runs/${GITHUB_RUN_ID}" )
+COMMIT_ORIGINAL_AUTHOR="${GITHUB_ACTOR} <${GITHUB_ACTOR}@users.noreply.github.com>"
+
+if [[ "$GITHUB_REF" =~ ^refs/heads/ ]]; then
+	BRANCH=${GITHUB_REF#refs/heads/}
+else
+	echo "Could not determine branch name from $GITHUB_REF"
+	exit 1
+fi
+
+# Install Yarn generally.
+echo "::group::Monorepo setup"
+yarn install
+echo "::endgroup::"
+
+echo "Cloning projects and pushing to Automattic mirror repos"
+EXIT=0
+
+# sync to read-only clones
+for project in projects/packages/* projects/plugins/*; do
+	PROJECT_DIR="${BASE}/${project}"
+	[[ -d "$PROJECT_DIR" ]] || continue # We are only interested in directories (i.e. projects)
+
+	printf "\n\n\e[7m Project: %s \e[0m\n" "$project"
+
+	cd "${PROJECT_DIR}"
+
+	# Read mirror repo from composer.json, if it exists
+	if [[ -f "composer.json" ]]; then
+		COMPOSER_JSON_EXISTED=true
+		GIT_SLUG=$(jq -r '.extra["mirror-repo"] // ( .name | sub( "^automattic/"; "Automattic/" ) )' composer.json)
```

Comment:
I really dig this. I'd want to iterate on this to include defining a production and development build step in `package.json` so we can abstract that out for other projects to use below -- but this method of doing it is slick. Just need to be sure to document it when we document the monorepo.